src/main/resources/xlr_xldeploy/XLDeployClient.py

```diff
@@ -333,6 +333,20 @@ def check_ci_exist(self, ci_id, throw_on_fail=False):
             raise Exception("CI with id [%s] does not exist." % ci_id)
         return False
 
+    def create_folder_tree(self, folder_id, folder_type):
+        folders = folder_id.split("/")
+        folderCreate = '%s' % folder_type
+        for folder in folders:
+            folderCreate += "/" + folder
+            # Start the request to XLD
+            body = '<list><core.Directory id="' + folderCreate + '"/></list>'
+            create_task = "/deployit/repository/cis"
+            response = self.http_request.put(create_task, body, contentType='application/xml')
```

Comment:
This is because the `create_directory` method uses a `POST`, which will fail if a directory already exists. This will be a problem when creating a subfolder to a folder that already exists.
The PUT method used here will either "update" the folder or create it and not fail if a directory already exists.